subtask_9.2/src/ThreadDemo.java

```diff
@@ -0,0 +1,23 @@
+import java.util.concurrent.Semaphore;
+
+public class ThreadDemo implements Runnable{
+
+    private Semaphore semaphore;
+
+    public ThreadDemo() {
+        this.semaphore = new Semaphore(1);
```

Comment:
That's not gonna work. Test it 
Read more carefully about semaphores and hot to use them.